client/app/lib/redux/modules/bongo.coffee

```diff
@@ -0,0 +1,77 @@
+_ = require 'lodash'
+immutable = require 'app/util/immutable'
+
+{ makeNamespace, expandActionType } = require 'app/redux/helper'
+
+withNamespace = makeNamespace 'koding', 'bongo'
+
+LOAD = expandActionType withNamespace 'LOAD'
+REMOVE = expandActionType withNamespace 'REMOVE'
+
+reducer = (state = immutable({}), action) ->
+
+  switch action.type
+    when LOAD.SUCCESS
+      { result } = action
+      result = [result]  unless Array.isArray result
+      result.forEach (res) ->
+        state = state.set res.constructor.name, immutable {}  unless state[res.constructor.name]
+        state = state.update res.constructor.name, (collection) ->
+          collection = collection.set res._id, immutable res
+          return collection
```

Comment:
Yep, probably i made a debugging session to see the result of `collection` and then forgot to remove the explicit return. Will fix.